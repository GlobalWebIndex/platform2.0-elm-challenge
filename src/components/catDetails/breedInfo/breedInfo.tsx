import React from 'react';
import {Link} from 'react-router-dom';
import {type Cat} from 'cat-lover/types/cats';

type BreedInfoProps = {
  cat: Cat;
  className: string;
};

export const BreedInfo = ({cat, className}: BreedInfoProps): JSX.Element => (
  <div className={className}>
    <h2>Breeds</h2>
    <ul>
      {cat.breeds?.map(breed => (
        <li key={breed.id}>
          <h3>{breed.name}</h3>
          <ul>
            <li>
              <strong>Origin:</strong> {breed.origin}
            </li>
            <li>
              <strong>Temperament:</strong> {breed.temperament}
            </li>
            <li>
              <strong>Life span:</strong> {breed.life_span} years
            </li>
            <li>
              <strong>Wikipedia:</strong>{' '}
              <a href={breed.wikipedia_url} rel="noreferrer" target="_blank">
                {breed.wikipedia_url}
              </a>
            </li>
            <li>
              <Link to={`/breeds/${breed.id}`}>Visit the breed page</Link>
            </li>
          </ul>
        </li>
      ))}
    </ul>
  </div>
);
