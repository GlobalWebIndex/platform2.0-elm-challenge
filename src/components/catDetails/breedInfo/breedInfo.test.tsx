import React from 'react';
import {MemoryRouter} from 'react-router-dom';
import {render} from '@testing-library/react';
import type {Cat} from 'cat-lover/types/cats';
import {BreedInfo} from './breedInfo';

describe('<BreedInfo />', () => {
  const mockCat: Cat = {
    breeds: [
      {
        country_code: '',
        country_codes: '',
        id: '1',
        life_span: '10-12',
        name: 'Breed One',
        origin: 'Country One',
        temperament: 'Calm',
        weight: {
          imperial: '',
          metric: ''
        },
        wikipedia_url: 'https://en.wikipedia.org/wiki/Breed_One'
      }
    ],
    height: 100,
    id: '1',
    url: 'https://example.com/cat.jpg',
    width: 200
  };
  const props = {
    cat: mockCat,
    className: 'testClass'
  };

  it('should render breed information', () => {
    const {container} = render(
      <MemoryRouter initialEntries={['/cats/1']}>
        <BreedInfo {...props} />
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
  });

  describe('when the cat has multiple breeds', () => {
    const multipleBreedsCat: Cat = {
      ...mockCat,
      breeds: [
        ...(mockCat.breeds as NonNullable<Cat['breeds']>),
        {
          country_code: '',
          country_codes: '',
          id: '2',
          life_span: '12-14',
          name: 'Breed Two',
          origin: 'Country Two',
          temperament: 'Active',
          weight: {
            imperial: '',
            metric: ''
          },
          wikipedia_url: 'https://en.wikipedia.org/wiki/Breed_Two'
        }
      ]
    };
    const multipleBreedsProps = {
      ...props,
      cat: multipleBreedsCat
    };

    it('should render with multiple breeds', () => {
      const {container} = render(
        <MemoryRouter initialEntries={['/cats/1']}>
          <BreedInfo {...multipleBreedsProps} />
        </MemoryRouter>
      );

      expect(container).toMatchSnapshot();
    });
  });

  it('should render link that navigates tp the "breed details" screen', () => {
    const {getByText} = render(
      <MemoryRouter initialEntries={['/cats/1']}>
        <BreedInfo {...props} />
      </MemoryRouter>
    );
    const moreDetailsLink = getByText('Visit the breed page');

    expect(moreDetailsLink).toBeInTheDocument();
    expect(moreDetailsLink).toHaveAttribute('href', '/breeds/1');
  });
});
