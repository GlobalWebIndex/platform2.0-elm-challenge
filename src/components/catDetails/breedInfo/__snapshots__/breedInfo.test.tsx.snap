// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BreedInfo /> should render breed information 1`] = `
<div>
  <div
    class="testClass"
  >
    <h2>
      Breeds
    </h2>
    <ul>
      <li>
        <h3>
          Breed One
        </h3>
        <ul>
          <li>
            <strong>
              Origin:
            </strong>
             
            Country One
          </li>
          <li>
            <strong>
              Temperament:
            </strong>
             
            Calm
          </li>
          <li>
            <strong>
              Life span:
            </strong>
             
            10-12
             years
          </li>
          <li>
            <strong>
              Wikipedia:
            </strong>
             
            <a
              href="https://en.wikipedia.org/wiki/Breed_One"
              rel="noreferrer"
              target="_blank"
            >
              https://en.wikipedia.org/wiki/Breed_One
            </a>
          </li>
          <li>
            <a
              href="/breeds/1"
            >
              Visit the breed page
            </a>
          </li>
        </ul>
      </li>
    </ul>
  </div>
</div>
`;

exports[`<BreedInfo /> when the cat has multiple breeds should render with multiple breeds 1`] = `
<div>
  <div
    class="testClass"
  >
    <h2>
      Breeds
    </h2>
    <ul>
      <li>
        <h3>
          Breed One
        </h3>
        <ul>
          <li>
            <strong>
              Origin:
            </strong>
             
            Country One
          </li>
          <li>
            <strong>
              Temperament:
            </strong>
             
            Calm
          </li>
          <li>
            <strong>
              Life span:
            </strong>
             
            10-12
             years
          </li>
          <li>
            <strong>
              Wikipedia:
            </strong>
             
            <a
              href="https://en.wikipedia.org/wiki/Breed_One"
              rel="noreferrer"
              target="_blank"
            >
              https://en.wikipedia.org/wiki/Breed_One
            </a>
          </li>
          <li>
            <a
              href="/breeds/1"
            >
              Visit the breed page
            </a>
          </li>
        </ul>
      </li>
      <li>
        <h3>
          Breed Two
        </h3>
        <ul>
          <li>
            <strong>
              Origin:
            </strong>
             
            Country Two
          </li>
          <li>
            <strong>
              Temperament:
            </strong>
             
            Active
          </li>
          <li>
            <strong>
              Life span:
            </strong>
             
            12-14
             years
          </li>
          <li>
            <strong>
              Wikipedia:
            </strong>
             
            <a
              href="https://en.wikipedia.org/wiki/Breed_Two"
              rel="noreferrer"
              target="_blank"
            >
              https://en.wikipedia.org/wiki/Breed_Two
            </a>
          </li>
          <li>
            <a
              href="/breeds/2"
            >
              Visit the breed page
            </a>
          </li>
        </ul>
      </li>
    </ul>
  </div>
</div>
`;
